pkg/prefetch/prefetch.go

```diff
@@ -34,14 +44,88 @@ func (p *prefetchInfo) SetPrefetchFiles(body []byte) error {
 	}
 	for _, item := range prefetchMsg {
 		image := item["image"]
-		prefetchfiles := item["prefetch"]
+		url := item["prefetch"]
+		prefetchfiles, err := p.getprefetchConfigFile(url, image)
+		if err != nil {
+			return err
+		}
 		p.prefetchMap[image] = prefetchfiles
 	}
 
 	log.L.Infof("received prefetch list from nri plugin: %v ", p.prefetchMap)
 	return nil
 }
 
+// Obtain the prefetch files through the URL and transfer them to a local txt file.
+func (p *prefetchInfo) getprefetchConfigFile(url, image string) (string, error) {
+	resp, err := http.Get(url)
+	if err != nil {
+		return "", err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		errMsg := fmt.Sprintf("HTTP response status code is not 200 OK: %s", resp.Status)
+		return "", errors.New(errMsg)
+	}
+	dirPath := prefetchlistPath
+	if _, err = os.Stat(dirPath); os.IsNotExist(err) {
+		err = os.MkdirAll(dirPath, 0755)
+		if err != nil {
+			return "", err
+		}
+	}
+	replaced := strings.ReplaceAll(image, "/", "@")
+	filePath := dirPath + "/" + replaced
+	file, err := os.Create(filePath)
+	if err != nil {
+		return "", err
+	}
+
+	defer func() {
+		if err = file.Close(); err != nil {
+			log.L.Fatalf("Error while closing the file: %v", err)
+		}
+	}()
+	var htmlBuffer bytes.Buffer
+
+	tokenizer := html.NewTokenizer(resp.Body)
+	inBody := false
+
+	for {
```

Comment:
I don't understand why we need this loop. May have a more concise implementation? 
Could you take a look at this? cc @imeoer

Reply:
Yes, we can't assume the server side returns such html format.